src/views/primary-auth/PrimaryAuthForm.js

```diff
@@ -43,10 +44,23 @@ define([
             password: this.model.get('password')
           });
         }
-        this.model.save();
+        if (this.settings.get('features.deviceFingerprinting')) {
+          var self = this;
+          DeviceFingerprint.generateDeviceFingerprint(this.settings.get('baseUrl'))
+          .then(function (fingerprint) {
+            self.model.set('deviceFingerprint', fingerprint);
+            self.model.save();
+          })
+          .fail(function () {
```

Comment:
Use .fin to do something in both cases (fail and then) rather than duplicating the logic